src/v/cluster/node/local_monitor.cc

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "cluster/node/local_monitor.h"
+
+#include "cluster/node/types.h"
+#include "config/node_config.h"
+#include "model/timestamp.h"
+#include "version.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/do_with.hh>
+#include <seastar/core/file-types.hh>
+#include <seastar/core/file.hh>
+#include <seastar/core/future.hh>
+#include <seastar/core/gate.hh>
+#include <seastar/core/iostream.hh>
+#include <seastar/core/reactor.hh>
+#include <seastar/core/scheduling.hh>
+
+#include <chrono>
+#include <seastarx.h>
+
+namespace cluster::node {
+
+// TODO make async
+void local_monitor::update_state() {
+    // XXX AJF review question: do we need an abort source here?
```

Comment:
what are you thinking might need to be aborted?

Reply:
I don't think it is needed but wanted to be sure. Tick timer arm() is already covered with abort source, and this codepath is protected with a gate, so I think it is ok.

Reply:
normally a gate is used to synchronous fibers during shutdown, and an abort source is used to trigger the action of starting to shutdown fibers. sometimes they blend together a little bit w.r.t. signaling that shutdown should start.